File: judge/java/tests/second_largest_number.java
package tests;
import java.util.*;
import java.lang.*;
import java.io.*;
import judge.*;
import datastruct.*;

public class second_largest_number {
    public static int num_test = 75;
    public static int[][] in_0;
    public static int[][] in_org_0;
    public static int[] out;


    public static void load_test() {
        File fil = new File("judge/tests/second-largest-number.txt");
        FileReader inputFil = null;
        try {
            inputFil = new FileReader(fil);
        } catch (FileNotFoundException e) {
            e.printStackTrace();
        }
        BufferedReader in = new BufferedReader(inputFil);
        try {
            in_0 = common.read_int_matrix(in);
            in_org_0 = test_common.copy(in_0);
            out = common.read_int_array(in);
        } catch (IOException e) {
            e.printStackTrace();
        }
    }

    public static void judge() {
        load_test();
        Solution s = new Solution();

        long startTime = System.currentTimeMillis();

        for(int i = 0; i < num_test; ++i) {
            int answer = s.second_largest(in_0[i]);
            if(answer != out[i]) {
                System.out.printf("%d / %d;", i+1, num_test);
                String outs = common.to_string(second_largest_number.in_org_0[i]);
                System.out.print(outs + ";");
                System.out.print(common.to_string(answer) + ";");
                System.out.println(common.to_string(out[i]));
                return;
            }
        }

        long estimatedTime = System.currentTimeMillis() - startTime;
        System.out.print("Accepted;");
        System.out.println(estimatedTime);
    }
}

File: judge/scala/tests/second_largest_number.scala
package test
import scala.io.Source
import judge.common
import judge.Solution

object second_largest_number {
    val num_test = 75;
    var in_0 = List[List[Int]]();
    var out = List[Int]();


    def load_test() = {
        val in = Source.fromFile("./judge/tests/second-largest-number.txt").getLines;
        in_0 = common.read_int_matrix(in);
        out = common.read_int_array(in);
    }

    def judge(): Int = {
        load_test();
        common.capture_stdout();

        val startTime = System.currentTimeMillis();
        var i = 0;

        while(i < num_test) {
            printf("Testing case #%d\n", i+1);
            val answer = Solution.second_largest(in_0(i));
            if(answer != out(i)) {
                common.release_stdout();
                printf("%d / %d;", i+1, num_test);
                var outs = common.to_string(second_largest_number.in_0(i));
                print(outs + ";");
                print(common.to_string(answer) + ";");
                println(common.to_string(out(i)));
                return 1;
            }
            i += 1;
        }

        common.release_stdout();
        val estimatedTime = System.currentTimeMillis - startTime;
        print("Accepted;");
        println(estimatedTime);
        return 0;
    }
}
